packages/astro/src/core/config/schema.ts

```diff
@@ -326,6 +325,11 @@ export function createRelativeSchema(cmd: string, fileProtocolRoot: URL) {
 		) {
 			config.build.client = new URL('./dist/client/', config.outDir);
 		}
+		if(trimSlashes(config.base).length && config.trailingSlash === 'never') {
+      config.base = prependForwardSlash(trimSlashes(config.base));
+    } else {
+      config.base = prependForwardSlash(appendForwardSlash(trimSlashes(config.base)));
+    }
```

Comment:
nit: it would be nice if `trimSlashes(config.base)` is assigned as a variable so it can be reuse